lib/db.js

```diff
@@ -379,7 +379,9 @@ Db.prototype.collection = function(name, options, callback) {
   options.promiseLibrary = this.s.promiseLibrary;
 
   // If we have not set a collection level readConcern set the db level one
-  options.readConcern = options.readConcern || this.s.readConcern;
+  options.readConcern = options.readConcern
+    ? new ReadConcern(options.readConcern.level)
```

Comment:
We should be using the inheritance here! `options.readConcern = this.readConcern` will do this resolution for you

Reply:
I'm not sure I follow. Wouldn't `options.readConcern = this.readConcern` disregard `options.readConcern` if it was passed to the operation?

Reply:
Oops, I misread this. Follow-up questions though: if `options.readConcern` is present, isn't it guaranteed to be a `ReadConcern` object already? Why create a new object? Secondly, shouldn't we use `this.readConcern`, so you potentially get a inherited values (when we eventually allow inheritance from `MongoClient`)?

Reply:
Someone could do `const newCollection = db.collection('test', { readConcern: { level: 'local' } }, callback);` This means that `options.readConcern` is not guaranteed to be a `ReadConcern` object already. We should be using `this.readConcern` so I'll fix that.